frontend/test/pytest/test_jax_integration.py

```diff
@@ -0,0 +1,253 @@
+# Copyright 2023 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Test QJIT compatibility with JAX transformations such as jax.jit and jax.grad."""
+
+from functools import partial
+
+import jax
+import jax.numpy as jnp
+import pennylane as qml
+import pytest
+
+from catalyst import measure, qjit
+
+
+class TestJAXJIT:
+    """Test QJIT compatibility with JAX compilation."""
+
+    def test_simple_circuit(self):
+        """Test a basic use case of jax.jit on top of qjit."""
+
+        @qjit
+        @qml.qnode(qml.device("lightning.qubit", wires=2))
+        def circuit(x: jax.ShapedArray((3,), dtype=float)):
+            qml.RX(jnp.pi * x[0], wires=0)
+            qml.RY(x[1] ** 2, wires=0)
+            qml.RX(x[1] * x[2], wires=0)
+            return qml.expval(qml.PauliZ(0))
+
+        @jax.jit
+        def cost_fn(x):
+            result = circuit(x)
+            return jnp.cos(result) ** 2
+
+        result = cost_fn(jnp.array([0.1, 0.2, 0.3]))
+        assert jnp.allclose(result, 0.3573519347253657)
```

Comment:
@dime10 I recommend using analytic results for the comparison. E.g., for this circuit, and the circuits below, the analytic derivative should be pretty simple (a basic trig function).

We make a deliberate effort to do this in PL, because there were quite a few times bugs were not caught because 'magic numbers' used in test comparisons were computed incorrectly, whereas analytic results are easy to check

Reply:
You can pilfer the [pennylane jax tests](https://github.com/PennyLaneAI/pennylane/blob/master/tests/interfaces/test_jax_qnode.py) for some examples :)

Reply:
> We make a deliberate effort to do this in PL, because there were quite a few times bugs were not caught because 'magic numbers' used in test comparisons were computed incorrectly, whereas analytic results are easy to check

That's a fair point. My thinking was to follow your advice of not comparing against executions of other frameworks (e.g. comparing to the result produced by PennyLane every time), but a magic number might actually be worse 😅

Reply:
Yeah, we've had issues in the past with both comparisons to other frameworks _and_ to magic numbers 😅 Analytic comparisons though have worked really well.

Reply:
I removed all comparisons to magic numbers here: https://github.com/PennyLaneAI/catalyst/pull/96/commits/cd18477afd34914a5a70e93df00285078554b8cc